### ISSUE TITLE
Fixed packages in Dockerfile

### ISSUE THREAD

**Issue by svaningelgem** [azinchen/nordvpn#320]
Hi,

I'm wondering why you have fixed packages inside your Dockerfile. Why not simply let them float and let apk install the latest versions?
The reason I came onto this is that I was looking how you were updating these version numbers but didn't see any github workflows for it.

Is it because you want a reproducible build every time?
Or is it because you want to vet the version increases yourself?
Or still something else?

I'd like to learn more ;-)

Thanks

**Comment by azinchen** [comment]
Hi @svaningelgem ,

_I'm wondering why you have fixed packages inside your Dockerfile. Why not simply let them float and let apk install the latest versions?_
Long story short, check Dockerfile in Linter https://hadolint.github.io/hadolint/. There is uncompleted work in `implementlinter` branch where I tried to implement github workflow to check all code in linter.

_The reason I came onto this is that I was looking how you were updating these version numbers but didn't see any github workflows for it._
I found some github actions for automatic updating package versions, but didn't test it.

The advantage of fixed packages is automatic triggering Build and Deploy workflow after updating Dockerfile. Usually I update package versions after receiving security check reports from Snyk.
The disadvantage is broken builds when new version of package is introduced because old version is removed from Alpine Linux repositories.
